packages/web-components/fast-element/src/directives/node-observation.ts

```diff
@@ -55,7 +88,7 @@ export abstract class NodeObservationBehavior<T extends NodeBehaviorBehaviorOpti
             (x: Accessor) => x.name === name
         );
         this.source = source;
-        this.updateTarget(this.getNodes());
+        this.updateTarget(this.getNodes().filter(this.options.filter!));
```

Comment:
How often do you anticipate custom filters to not be provide? Would it make sense to skip the filter operation if no filter is provided?

Reply:
I'm honestly not sure. It might still be good to make the optimization though. Let me adjust that.